cmd/tailscale/cli/serve_dev.go

```diff
@@ -411,20 +417,12 @@ func (e *serveEnv) applyWebServe(sc *ipn.ServeConfig, dnsName string, srvPort ui
 			return errors.New("unable to serve; text cannot be an empty string")
 		}
 		h.Text = text
-	case isProxyTarget(target):
```

Comment:
Changing the case order simplified the logic and allowed us to only provide a port in `expandProxyTargetDev`. 

Now we check if it starts with `text:`, if it's an absolute path, and otherwise is a proxy target.